### PR TITLE
Wheeling port icons

### LINKED ISSUE #880: 1846 - upgrading G19/Wheeling should preserve both port icons
This was caused by this line, but I haven't looked at 1882 or that PR enough to know that line was fixing - https://github.com/tobymao/18xx/pull/826/files#diff-489105431840138935ce969dd570ab2cR127

Before the upgrade:
Screenshot from 2020-06-25 23-53-37

Current behavior:
Screenshot from 2020-06-25 23-51-57

Desired behavior:
Screenshot from 2020-06-20 15-38-49

### PR DESCRIPTION
**Pull request by michaeljb**
Fixes #880

### CODE REVIEW

File: lib/engine/game/base.rb [hunk 1]
```diff
@@ -227,7 +227,14 @@ def initialize(names, id: 0, actions: [], pin: nil, strict: false)
         @finished = false
         @log = []
         @actions = []
-        @names = names.freeze
+
+        @names =
```
Comment (tobymao) [inline lib/engine/game/base.rb:231]: let's not do this

Comment (michaeljb) [reply lib/engine/game/base.rb]: ok, reverted in cfca4a0e

File: lib/engine/hex.rb [hunk 2]
```diff
@@ -135,11 +135,10 @@ def lay(tile)
...
       end
 
-      @tile.icons.each do |icon|
-        next unless icon.sticky
-        next if tile.icons.any? { |x| icon.name == x.name }
-
-        new_icon = icon.dup
+      sticky_icon_names = @tile.icons.select(&:sticky).map(&:name)
+      new_icon_names = tile.icons.map(&:name)
+      (sticky_icon_names - new_icon_names).each do |icon_name|
```
Comment (tobymao) [inline lib/engine/hex.rb:140]: new_icons = tiles.icons.group_by(&:name)

@tile.icons.each do |icon|
  next unless icon.sticky
  next if new_icons[icon.name]&.pop  

  new_icon = icon.dup
  ...

Comment (michaeljb) [reply lib/engine/hex.rb]: cool, thanks :+1: fd2b7d8d

**Comment by tobymao** [comment]
lgtm  feel free to merge and deploy

**Pull request merged**